runtimes/clone/src/lib.rs

```diff
@@ -197,8 +196,8 @@ impl pallet_transaction_payment::Config for Runtime {
 }
 
 parameter_types! {
-	pub const ReservedXcmpWeight: Weight = constants::MAXIMUM_BLOCK_WEIGHT / 4;
-	pub const ReservedDmpWeight: Weight = constants::MAXIMUM_BLOCK_WEIGHT / 4;
+	pub const ReservedXcmpWeight: Weight = constants::MAXIMUM_BLOCK_WEIGHT.saturating_div(4);
```

Comment:
Can we not saturate here but fail since these are constants?

Reply:
We could but I wanted to stay in sync with Parity, see [Cumulus](https://github.com/paritytech/cumulus/blob/polkadot-v0.9.29/parachain-template/runtime/src/lib.rs#L217)

Reply:
Overflow doesn't panic tho do they? And these are not "consts" in a sense of rust "consts". This gets expanded to a function that returns `constants::MAXIMUM_BLOCK_WEIGHT / 4`. So the compiler might optimize that, but he doesn't need to.

The only overflow that can happen while using div is when dividing with `-1` anyways and that's not possible since this is a u64. So I actually might also remove the saturating_div.

Reply:
I suppose the reason for the saturated div here is that the compiler does not accept just dividing, e.g. it throws `cannot call non-const operator in constants` for `constants::MAXIMUM_BLOCK_WEIGHT / 4`.

I am unhappy with panicking here. But if you prefer, we can do that.

Reply:
BTW, when unwrapping or expecting the `checked_div`, the compiler also complains: 

```error: `Option::<T>::unwrap` is not yet stable as a const fn```

Do we value removing saturated division so much that we want to add `#![feature(const_option)]`?

Reply:
aha interesting! So it's actually a const function. Didn't look close enough then.

Reply:
It is a constant hence using a simple `/` should be enough, as any invalid case would be caught at compilation time (when compiled in release mode, any overflow or division by zero is also caught). A saturating division would, potentially, saturate without giving any notice. I don't see why we would want to replace a simple division with a saturating one. Or am I missing something else?

Reply:
I also don't see a strong case for the saturating_div here.

Reply:
> [... the compiler] throws `cannot call non-const operator in constants` for `constants::MAXIMUM_BLOCK_WEIGHT / 4`.